File: backend/internal/models/enrollment.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Enrollment struct {
	ID                   uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	UserID               uuid.UUID `gorm:"type:uuid;not null"`
	User                 User      `gorm:"foreignKey:UserID"`
	CourseID             uuid.UUID `gorm:"type:uuid;not null"`
	Course               Course    `gorm:"foreignKey:CourseID"`
	GatewayTransactionID string    `gorm:"type:varchar(255)"`
	Status               string    `gorm:"type:varchar(50);not null"` // Ex: active, desactive, expired, refunded, pending
	ExpiresAt            *time.Time
	CreatedAt            time.Time `gorm:"autoCreateTime"`
}

File: frontend/src/types/enrollment.ts
export interface EnrollmentCourse {
  ID: string
  Title: string
  Description: string
  ThumbnailURL: string
  Price: number
  IsPublished: boolean
}

export interface Enrollment {
  ID: string
  UserID: string
  CourseID: string
  Course: EnrollmentCourse
  GatewayTransactionID: string
  Status: string
  ExpiresAt: string | null
  CreatedAt: string
}
